### PR TITLE
9.10-HF/feature-NXP-24399-workmanager-shutdown-delay

### PR DESCRIPTION
**Pull request by pierre-gautier**

### CODE REVIEW

File: nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/WorkManagerImpl.java [hunk 1]
```diff
@@ -92,6 +93,8 @@
 
     protected static final String THREAD_PREFIX = "Nuxeo-Work-";
 
+    public static final String SHUTDOWN_DELAY_KEY = "nuxeo.work.shutdown.delay";
+
```
Comment (bdelbosc) [inline nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/WorkManagerImpl.java]: I would like the rational on not using time unit

**Comment by bdelbosc** [comment]
:+1:

**Comment by funshodavid** [comment]
:+1:

**Comment by kevinleturc** [comment]
👍

File: nuxeo-core/nuxeo-core-event/src/test/resources/test-shutdown-delay.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<component name="org.nuxeo.core.event.workmanagerimpl.shutdown.delay">
+  <extension target="org.nuxeo.runtime.ConfigurationService" point="configuration">
+    <property name="org.nuxeo.ecm.core.work.shutdown.delay">5000</property>
+  </extension>
```
Comment (atchertchian) [inline nuxeo-core/nuxeo-core-event/src/test/resources/test-shutdown-delay.xml]: IMHO this deserves documentation (sample: https://github.com/nuxeo/nuxeo/blob/release-9.10/nuxeo-services/nuxeo-platform-usermanager-core/src/main/resources/OSGI-INF/usermanager-properties.xml)

Comment (atchertchian) [reply nuxeo-core/nuxeo-core-event/src/test/resources/test-shutdown-delay.xml]: (discussed live, this is a test conf, will be documented later if still needed)

Comment (pierre-gautier) [reply nuxeo-core/nuxeo-core-event/src/test/resources/test-shutdown-delay.xml]: As it's a test contribution, it has been decided it wasn't necessary.

**Pull request merged**